common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/StorageManifestService.scala

```diff
@@ -105,65 +110,77 @@ object StorageManifestService extends Logging {
     *   - a file referenced by the fetch file, which should be in a different
     *     versioned directory under the same bag root
     *
-    * This function gets a map (bag name) -> (path), relative to the bag root.
+    * The map contains data
+    *
+    *   (bag path) -> (location, size if known)
     *
     */
-  private def createNamePathMap(
+  private def createPathLocationMap(
```

Comment:
I'm missing a fair bit of context, but I'm finding this method pretty nested and unwieldy (I know was already pretty big to start with!). Maybe it would be easier to read if some of the inner functionality was broken into a few smaller methods?

Reply:
Yep, fair comment, let me break it up a bit.